homeassistant/components/acmeda/sensor.py

```diff
@@ -0,0 +1,69 @@
+"""Support for Acmeda Roller Blind Batteries."""
+import asyncio
+
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL,
+    DEVICE_CLASS_BATTERY,
+    UNIT_PERCENTAGE,
+)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.icon import icon_for_battery_level
+
+from .base import AcmedaBase
+from .const import ACMEDA_HUB_UPDATE
+from .helpers import update_entities
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Acmeda Rollers from a config entry."""
+    update_lock = asyncio.Lock()
+    current = {}
+
+    async def async_update():
+        async with update_lock:
+            await update_entities(
+                hass, AcmedaBattery, config_entry, current, async_add_entities
+            )
+
+    async_dispatcher_connect(
+        hass, ACMEDA_HUB_UPDATE.format(config_entry.entry_id), async_update
+    )
+
+
+class AcmedaBattery(AcmedaBase):
+    """Representation of a Acmeda cover device."""
+
+    device_class = DEVICE_CLASS_BATTERY
+    unit_of_measurement = UNIT_PERCENTAGE
+
+    @property
+    def name(self):
+        """Return the name of roller."""
+        return f"{super().name} Battery"
+
+    @property
```

Comment:
Please remove `battery_level` and `device_state_attributes` as its already in the state.   We only need it once

Reply:
Done.